store/migration/dev/LATEST.sql

```diff
@@ -128,21 +128,24 @@ EXECUTE FUNCTION trigger_update_updated_ts();
 -- Policy
 -- policy stores the policies for each environment.
 -- Policies are associated with environments. Since we may have policies not associated with environment later, we name the table policy.
+CREATE TYPE resource_type AS ENUM ('WORKSPACE', 'ENVIRONMENT', 'PROJECT', 'INSTANCE', 'DATABASE');
+
 CREATE TABLE policy (
     id SERIAL PRIMARY KEY,
     row_status row_status NOT NULL DEFAULT 'NORMAL',
     creator_id INTEGER NOT NULL REFERENCES principal (id),
     created_ts BIGINT NOT NULL DEFAULT extract(epoch from now()),
     updater_id INTEGER NOT NULL REFERENCES principal (id),
     updated_ts BIGINT NOT NULL DEFAULT extract(epoch from now()),
-    environment_id INTEGER NOT NULL REFERENCES environment (id),
     type TEXT NOT NULL CHECK (type LIKE 'bb.policy.%'),
-    payload JSONB NOT NULL DEFAULT '{}'
+    payload JSONB NOT NULL DEFAULT '{}',
+    resource_type resource_type NOT NULL,
+    resource_id INTEGER NOT NULL
 );
 
-CREATE INDEX idx_policy_environment_id ON policy(environment_id);
+CREATE INDEX idx_policy_resource_type_resource_id ON policy(resource_type, resource_id);
```

Comment:
I guess this is not needed since we have a UNIQUE INDEX below covering it

Reply:
so true. Done.